webportal/webapp/src/app/account/loginsync/components/loginsync.component.ts

```diff
@@ -11,34 +12,73 @@ import { CognitoService } from '../../../../services/cognito.service';
 export class LoginSyncComponent implements OnInit {
   username: string;
   password: string;
-  linkSuccessful = true;
+  newPassword: string;
+  newPasswordConfirmation: string;
+  changeTemporaryPassword = false;
   errorMessage = '';
+  complexityErrorMessage: string = "The password does not match the complexity criteria.";
+  passwordMatchingErrorMessage: string = "Passwords must match.";
 
-  constructor(private loginSyncService: LoginSyncService, private cognitoService: CognitoService) { }
+  constructor(private loginSyncService: LoginSyncService, private cognitoService: CognitoService,  @Inject(WindowToken) private window: Window) { }
```

Comment:
This window token is how Angular suggests injecting `window` into components for testing.